src/cargo/ops/registry.rs

```diff
@@ -242,6 +242,20 @@ fn transmit(
                 .with_context(|| format!("failed to read `readme` file for package `{}`", pkg))
         })
         .transpose()?;
+
+    let readme_path_relative_to_repo = readme.as_ref().map(|readme| {
+        || -> Option<String> {
+            let repo = git2::Repository::discover(pkg.root()).ok()?;
```

Comment:
would this ever problematically discover too high up (eg maybe user made their whole root a git directory, and are working on a cargo project that isn't in git)? Is there some way to bound this discovery? Maybe we don't care so much for that case. It seems very contrived.